statsd/src/test/java/com/avast/metrics/statsd/StatsDTimerTest.java

```diff
@@ -0,0 +1,88 @@
+package com.avast.metrics.statsd;
+
+import com.timgroup.statsd.StatsDClient;
+import org.junit.Test;
+import org.mockito.Matchers;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+import static org.mockito.Mockito.*;
+
+public class StatsDTimerTest {
+
+    @Test
+    public void testCounts() throws Exception {
```

Comment:
this test seems a little bit crazy to me :) it is testing so many things that I would personally split it to more tests :)